plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryGenerateProguardUuidTask.kt

```diff
@@ -2,35 +2,50 @@ package io.sentry.android.gradle.tasks
 
 import io.sentry.android.gradle.util.info
 import java.util.UUID
+import org.gradle.api.DefaultTask
 import org.gradle.api.Project
 import org.gradle.api.file.Directory
+import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.file.RegularFile
 import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
 import org.gradle.api.tasks.TaskProvider
 
-abstract class SentryGenerateProguardUuidTask : DirectoryOutputTask() {
+abstract class SentryGenerateProguardUuidTask : DefaultTask() {
 
     init {
         outputs.upToDateWhen { false }
         description = "Generates a unique build ID to be used " +
             "when uploading the Sentry mapping file"
     }
 
+    @get:OutputDirectory
+    @get:Optional
+    abstract val output: DirectoryProperty
+
     @get:Internal
     val outputFile: Provider<RegularFile> get() = output.file(SENTRY_UUID_OUTPUT)
 
     @TaskAction
     fun generateProperties() {
-        val outputDir = output.get().asFile
-        outputDir.mkdirs()
+        val outputDir = output.orNull?.asFile
+
+        if (outputDir != null) {
```

Comment:
Does this cause problems where the file is no longer available for other tasks that rely on it, like `SentryUploadProguardMappingsTask`?